File: benchmark/c/cpw/number_unique_rectangles_formed_using_n_unit_squares/number_unique_rectangles_formed_using_n_unit_squares.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int number_unique_rectangles_formed_using_n_unit_squares ( int n ) ;
int number_unique_rectangles_formed_using_n_unit_squares ( int n ) {
  int ans = 0;
  for ( int length = 1;
  length <= sqrt ( n );
  ++ length ) for ( int height = length;
  height * length <= n;
  ++ height ) ans ++;
  return ans;
}

File: translation_result/else_bms/success/c/cpw/number_unique_rectangles_formed_using_n_unit_squares/gemini/Hinted/number_unique_rectangles_formed_using_n_unit_squares.rs

fn number_unique_rectangles_formed_using_n_unit_squares(n: i32) -> i32 {
    let mut ans = 0;
    for length in 1..=(n as f64).sqrt() as i32 {
        for height in length..=(n / length) {
            ans += 1;
        }
    }
    ans
}
